ni/nimble
# Fix bug where table rows are not rendered correctly when the table is reattached to the DOM

# Pull Request

## 🤨 Rationale

Fixes #1753 

## 👩‍💻 Implementation

The bug in the existing code is that the table's viewport always reset to a scroll position of 0 when reconnected, but the virutalizer maintained its state of being scrolled to a certain position. To fix this problem, I updated the table's viewport scroll position to match the virtualizer's scroll position once the state of the virtualizer has been updated upon reconnection of the `nimble-table` element.

## 🧪 Testing

- Manually tested
- Wrote some new unit tests that verify that the scroll position and rendered rows are as expected when the table is disconnected and reconnected. This includes tests that update the data while the table is not connected.

## ✅ Checklist
